File: sol/442.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        res = []
        for i in range(len(nums)):
            a = abs(nums[i])
            if nums[a - 1] > 0:
                nums[a - 1] *= -1
            else:
                res.append(a)
        return res

File: sol/442.cpp
struct Solution {
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> res;
        for (int i = 0; i < nums.size(); ++i){
            int a = abs(nums[i]);
            if (nums[a - 1] > 0){
                nums[a - 1] = -nums[a - 1];
            } else {
                res.push_back(a);
            }
        }
        return res;
    }
};
